ifsbench/data/namelisthandler.py

```diff
@@ -0,0 +1,167 @@
+# (C) Copyright 2020- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from enum import auto, Enum
+import pathlib
+
+import f90nml
+
+from .datahandler import DataHandler
+from ..logging import debug, info
+
+
+__all__ = ['NamelistOverride', 'NamelistHandler']
+
+
+class NamelistOverride:
+    """
+    Specify changes that will be applied to a namelist.
+    """
+    class NamelistOperation(Enum):
+        SET = auto()
+        APPEND = auto()
+        DELETE = auto()
+
+    def __init__(self, key, mode, value=None):
+        """
+        Parameters
+        ----------
+        key: str or iterable of str
+            The namelist entry that will be modified. Can be either a string
+            where '/' separates the namelist name and the entry key or an iterable
+            of strings of length two.
+
+        mode: NamelistOverride.NamelistOperation
+            What kind of operation is specified. Can be
+                * Set a certain entry.
+                * Append to an array entry.
+                * Delete an entry.
+
+        value:
+            The value that is set (SET operation) or appended (APPEND).
+        """
+
+        if isinstance(key, str):
+            self._keys = key.split('/')
+        else:
+            self._keys = tuple(key)
+
+        if len(self._keys) != 2:
+            raise ValueError("The key object must be of length two.")
+
+        self._mode = mode
+        self._value = value
+
+        if self._value is None:
+            if self._mode in (self.NamelistOperation.SET, self.NamelistOperation.APPEND):
+                raise ValueError("The new value must not be None!")
+
+    def apply(self, namelist):
+        """
+        Apply the stored changes to a namelist.
+
+        Parameters
+        ----------
+        namelist: f90nml.Namelist
+            The namelist to which the changes are applied.
+        """
+
+        if self._keys[0] not in namelist:
+            if self._mode == self.NamelistOperation.DELETE:
+                return
+
+            namelist[self._keys[0]] = {}
+
+        namelist = namelist[self._keys[0]]
+        key = self._keys[-1]
+
+        if self._mode == self.NamelistOperation.SET:
+            debug(f"Set namelist entry {str(self._keys)} = {str(self._value)}.")
```

Comment:
[Purely FYI, no action required] The following should be a string conversion shorthand for f-strings

```suggestion
            debug(f"Set namelist entry {self._keys!s} = {self._value!s}.")
```